In [1]:
import csv
# Import guns.csv
f = open("guns.csv")
data= list(csv.reader(f))
# Let's look at the data
print(data[0:5])


[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


In [2]:
# Init count
count = 0
#Loop thru data
for row in data:
    # Increment counter
    count += 1
    # If it is the first line, write to header and remove from data
    if count == 1:
        headers= row
        data.remove(row)
        
print(headers)
print(data[0:5])

        

['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']
[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


In [3]:
# Create dictionary called year_counts
year_counts= {}
# Loop thru data
for row in data:
    # Identify year
    year = row[1]
    # See if year exists in year_counts
    if year in year_counts:
        # If so, increment counter for that year
        year_counts[year] = year_counts[year] + 1
    else:
        # Otherwise create an entry for that year with a count of 1
        year_counts[year] = 1
# Display Year Counts
print(year_counts)
    
    



{'2012': 33563, '2014': 33599, '2013': 33636}


In [4]:
import datetime

# Init Dates list and date_counts dictionary
dates= []
date_counts= {}

# Loop thru Data
for row in data:
    # Create a date from year and month, hardcode day as 1
    date_to_append = datetime.datetime(year=int(row[1]), month=int(row[2]), day=1)
    # Append newly created date to dates list
    dates.append(date_to_append)

print(dates[0:5])

# Loop thru dates list
for row in dates:
    # Write the row to date_to_check variable
    date_to_check = row
    # Check if date exists in date_counts, if so increment counter, otherwise add with count of 1
    if date_to_check in date_counts:
        date_counts[date_to_check] = date_counts[date_to_check] + 1
    else:
        date_counts[date_to_check] = 1
    
print(date_counts) 

[datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0)]
{datetime.datetime(2013, 6, 1, 0, 0): 2920, datetime.datetime(2012, 6, 1, 0, 0): 2826, datetime.datetime(2012, 4, 1, 0, 0): 2795, datetime.datetime(2014, 1, 1, 0, 0): 2651, datetime.datetime(2013, 10, 1, 0, 0): 2808, datetime.datetime(2013, 12, 1, 0, 0): 2765, datetime.datetime(2012, 7, 1, 0, 0): 3026, datetime.datetime(2014, 4, 1, 0, 0): 2862, datetime.datetime(2012, 8, 1, 0, 0): 2954, datetime.datetime(2014, 10, 1, 0, 0): 2865, datetime.datetime(2012, 5, 1, 0, 0): 2999, datetime.datetime(2014, 8, 1, 0, 0): 2970, datetime.datetime(2014, 5, 1, 0, 0): 2864, datetime.datetime(2012, 3, 1, 0, 0): 2743, datetime.datetime(2013, 4, 1, 0, 0): 2798, datetime.datetime(2013, 3, 1, 0, 0): 2862, datetime.datetime(2013, 7, 1, 0, 0): 3079, datetime.datetime(2013, 11, 1, 0, 0): 2758, datetime.datetime(2012, 11, 1, 0, 0

In [5]:
# Init sex_counts and race_counts
sex_counts = {}
race_counts = {}

# Loop thru data
for row in data:
    # Determine sex and race
    sex = row[5]
    race = row[7]
    # See if sex exists in sex_counts
    if sex in sex_counts:
        # If so, increment counter for that sex
        sex_counts[sex] = sex_counts[sex] + 1
    else:
        # Otherwise create an entry for that sex with a count of 1
        sex_counts[sex] = 1
    # See if sex exists in sex_counts
    if race in race_counts:
        # If so, increment counter for that race
        race_counts[race] = race_counts[race] + 1
    else:
        # Otherwise create an entry for that race with a count of 1
        race_counts[race] = 1       
        
# Display the dictionaries
print(sex_counts)
print(race_counts)



{'M': 86349, 'F': 14449}
{'White': 66237, 'Black': 23296, 'Asian/Pacific Islander': 1326, 'Hispanic': 9022, 'Native American/Native Alaskan': 917}


Gun Deaths

Number of Deaths:
This file covers the years of 2012-2014. During this time the # of gun deaths did not change significantly, with the difference between the lowest and highest # of gun deaths being 93 deaths.

2012 had 33563 gun deaths
2013 had 33636 gun deaths
2014 had 33599 gun deaths

Sex:
Males are more likely to die from guns then females. 

Race:
Native American/Native Alaskans had the lowest number of deaths, while Blacks had the highest number of deaths.

Black: 23296
White: 66237
Hispanic: 9022
Asian/Pacific Islander: 1326
Native American/Native Alaskan: 917

For further evaluation:

For race, we need to consider the percentage of the population that is made up of the race. The numbers are most informational if you know what percentage of that race was killed by guns in that time period. Comparing Black and .Native American/Native Alaskan deaths mean nothing if Native American/Native Alaskan is a small percentage of the overall population.

We have not yet considered intent, police, or location.


In [6]:
# Import census.csv
f = open("census.csv")
census= list(csv.reader(f))
print(census)


[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [7]:
# Create manual mapping
mapping = {
    "Asian/Pacific Islander": 15159516 + 674625,
    "Native American/Native Alaskan": 3739506,
    "Black": 40250635,
    "Hispanic": 44618105,
    "White": 197318956
}

# Init race_per_hundredk dictionary
race_per_hundredk = {}

# Loop thru race_counts
for key,value in race_counts.items():
    #Divide the value associated with the key in race_counts by the 
    #value associated with the key in mapping. Then multiply by 100000
    race_per_hundredk[key] = (value/mapping[key])*100000
    
print(race_per_hundredk)


{'White': 33.56849303419181, 'Black': 57.8773477735196, 'Asian/Pacific Islander': 8.374309664161762, 'Hispanic': 20.220491210910907, 'Native American/Native Alaskan': 24.521955573811088}


In [13]:
intents = []
races = []
# Loop thru data and gather intents and races
for item in data:
    intents.append(item[3])
    races.append(item[7])

homicide_race_per_hundredk = {}

# Enumerate races
for i, race in enumerate(races):
    # If race does not exist in dictionary, add it
    if race not in homicide_race_per_hundredk:
        homicide_race_per_hundredk[race] = 0
    # If the gun death was homicide, increment the counter
    if intents[i] == "Homicide":
        homicide_race_per_hundredk[race] += 1


# Loop thru race_counts
for key,value in homicide_race_per_hundredk.items():
    #Divide the value associated with the key in homicide_race_per_hundredk by the 
    #value associated with the key in mapping. Then multiply by 100000
    homicide_race_per_hundredk[key] = (value/mapping[key])*100000  
    
print(homicide_race_per_hundredk)    
            

{'White': 4.6356417981453335, 'Black': 48.471284987180944, 'Asian/Pacific Islander': 3.530346230970155, 'Hispanic': 12.627161104219914, 'Native American/Native Alaskan': 8.717729026240365}


Taking intent and percent of population changes our findings a bit. For example, Native Amerian/Native Alaskan had the lowest # of deaths, but when you take population and intent into the equation, Pacific Islander is lowest and Native Amerian/Native Alaskan is second lowest.